src/Avalonia.Controls/Platform/IWindowBaseImpl.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using Avalonia.Controls;
+
+namespace Avalonia.Platform
+{
+    public interface IWindowBaseImpl : ITopLevelImpl
+    {
+        /// <summary>
+        /// Shows the toplevel.
+        /// </summary>
+        void Show();
+
+        /// <summary>
+        /// Hides the window.
+        /// </summary>
+        void Hide();
+
+        /// <summary>
+        /// Starts moving a window with left button being held. Should be called from left mouse button press event handler.
+        /// </summary>
+        void BeginMoveDrag();
+
+        /// <summary>
+        /// Starts resizing a window. This function is used if an application has window resizing controls. 
+        /// Should be called from left mouse button press event handler
+        /// </summary>
+        void BeginResizeDrag(WindowEdge edge);
+
+        /// <summary>
+        /// Gets position of the window relatively to the screen
+        /// </summary>
+        Point Position { get; set; }
+        
+        /// <summary>
+        /// Gets or sets a method called when the window's position changes.
+        /// </summary>
+        Action<Point> PositionChanged { get; set; }
+
+        /// <summary>
+        /// Activates the window.
+        /// </summary>
+        void Activate();
+
+        /// <summary>
+        /// Gets or sets a method called when the window is deactivated (loses focus).
+        /// </summary>
+        Action Deactivated { get; set; }
+
+        /// <summary>
+        /// Gets or sets a method called when the window is activated (receives focus).
+        /// </summary>
+        Action Activated { get; set; }
+
+        /// <summary>
+        /// Gets the platform window handle.
+        /// </summary>
+        IPlatformHandle Handle { get; }
+        
+        /// <summary>
+        /// Gets the maximum size of a window on the system.
+        /// </summary>
+        Size MaxClientSize { get; }
+
+        /// <summary>
+        /// Gets the client size of the toplevel.
+        /// </summary>
+        void Resize(Size size);
```

Comment:
Needs a doc comment for the size parameter. Also maybe call this `clientSize` just to emphasize the fact it's the client size, not the size including borders.